#!/bin/sh /usr/share/dpatch/dpatch-run
##
## 99_debian_alsaconf_ignoreinstall.dpatch by Thomas Hood
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Use modprobe --ignore-install when modprobing legacy card drivers with args

@DPATCH@
diff -urNad alsa-utils-1.0.9/alsaconf/alsaconf.in /tmp/dpep.CMl3CX/alsa-utils-1.0.9/alsaconf/alsaconf.in
--- alsa-utils-1.0.9/alsaconf/alsaconf.in	2005-03-25 15:05:41.041891852 +0100
+++ /tmp/dpep.CMl3CX/alsa-utils-1.0.9/alsaconf/alsaconf.in	2005-03-25 15:08:01.589572000 +0100
@@ -820,7 +820,11 @@
 # return 0 - OK, 1 - NG, 2 - not working (irq/dma problem)
 ac_try_load () {
     test -n "$LOGFILE" && echo "$1 ${*:2}" >> "$LOGFILE"
+if [ "$kernel" = old ] ; then
     /sbin/modprobe snd-$1 ${*:2} >/dev/null 2>&1
+else
+    /sbin/modprobe --ignore-install snd-$1 ${*:2} >/dev/null 2>&1
+fi
     if $lsmod | grep -q -E '^(snd-|snd_)'$1' '; then
 	: ;
     else
@@ -859,7 +863,11 @@
 # ac_try_capture card duplex opts
 ac_try_capture () {
     test -n "$LOGFILE" && echo "$1 ${*:2}" >> "$LOGFILE"
+if [ "$kernel" = old ] ; then
     /sbin/modprobe snd-$1 ${*:3} >/dev/null 2>&1
+else
+    /sbin/modprobe --ignore-install snd-$1 ${*:3} >/dev/null 2>&1
+fi
     if $lsmod | grep -q -E '^(snd-|snd_)'$1' '; then
 	: ;
     else
